### PR TITLE
IGNITE-13772 Calcite, NPE on join.

### PR DESCRIPTION
**Pull request by zstan**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ExecutionTest.java [hunk 1]
```diff
@@ -66,6 +68,7 @@
  */
 @SuppressWarnings("TypeMayBeWeakened")
 @WithSystemProperty(key = "calcite.debug", value = "true")
+@RunWith(Parameterized.class)
```
Comment (tledkov-gridgain) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ExecutionTest.java:71]: Why you don't move this annotation to the base class `AbstractExecutionTest `?

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java [hunk 2]
```diff
@@ -69,6 +81,22 @@
...
+        params.add(new Object[]{true});
+        params.add(new Object[]{false});
+        params.add(new Object[]{null});
+
+        return params;
+    }
+
+    /** Execution direction. */
+    @Parameterized.Parameter
+    public static Boolean execDir;
```
Comment (tledkov-gridgain) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java:98]: May be enum usage will be more appropriate?

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java [hunk 3]
```diff
@@ -114,6 +142,59 @@ public void setup() throws Exception {
         }
     }
 
+    /** Task reordering executor. */
+    private static class IgniteRandomStripedThreadPoolExecutor extends IgniteStripedThreadPoolExecutor {
```
Comment (tledkov-gridgain) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java:146]: Naming suggestion: IgniteTestStripedThreadPoolExecutor. Not only random mode is supported

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java (hunk 2 above)
Comment (zstan) [reply modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java:98]: can`t use enum here, need to be iterable as i understand, does i missed something?

Comment (korlov42) [reply modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java:98]: @zstan please see my suggestion here: https://github.com/gridgain/apache-ignite/pull/259.

Also introducing params in the abstract class breaks ContinuousExecutionTest

Comment (zstan) [reply modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java]: thanks ! done.

**Comment by dspavlov** [comment]
This PR is associated with a resolved ticket. Closing the PR, please reopen it if you still need it.

**Pull request closed without merging**